util/test/sub_test

```diff
@@ -202,6 +202,105 @@ def LauncherTimeoutArgs(seconds):
               useLauncherTimeout)
 
 
+# Execute command.
+# Its stdout+stderr go to sub_test's stdout+stderr.
+# Return its exit code.
+def runAndShowOutput(command):
+    sys.stdout.flush()
+    p = subprocess.Popen(command)
+    rc = p.wait()
+    return rc
+
+# Execute command. Rc = Return code.
+# Gather its combined stdout+stderr.
+# Return (its exit code, the gathered output).
+def runAndReturnRcOutput(command):
+    p = subprocess.Popen(command,
+                         stdout=subprocess.PIPE,
+                         stderr=subprocess.STDOUT)
+    output = p.communicate()[0]
+    return p.returncode, output
+
+# Execute command. EIEO = Error If Error Output.
+# Gather *separately* its stdout and stderr.
+# If stderr is non-empty or the exit code is non-0, report an error.
+# If errfatal, quit with Fatal(). Otherwise, 
+# return the return code (at least 1 if got stderr) and command's stdout.
+def runAndReturnRcOutputEIEO(command, errfatal=False):
+    p = subprocess.Popen(command,
+                         stdout=subprocess.PIPE,
+                         stderr=subprocess.PIPE)
+    output = p.communicate()
+    errmsg = None
+    rc = 0
+
+    if output[1]:
+        errmsg = "unexpected stderr output while executing {0} (rc={1})".format(' '.join(command), p.returncode)
+        rc = 1
+        if errfatal:
+            sys.stdout.write("While executing {0}:\n".format(' '.join(command)))
+            sys.stdout.write("stdout=\n{0}stderr=\n{1}".format(trim_output(output[0]), trim_output(output[1])))
+        else:
+            sys.stdout.write("[Error: {0}]\n".format(errmsg))
+            sys.stdout.write("The stderr output was:\n{0}".format(trim_output(output[1])))
+    elif p.returncode:
+        errmsg = "rc={0} while executing {1}".format(p.returncode, ' '.join(command))
+        if not errfatal:
+            sys.stdout.write("[Error: {0}]\n".format(errmsg))
+        rc = p.returncode
+
+    if errfatal and errmsg:
+        Fatal(errmsg)
+
+    return rc, output[0]
+
+# Same as runAndReturnRcOutputEIEO() except drop return code.
+def runAndReturnOutputEIEO(command, errfatal=False):
+    rc, output = runAndReturnRcOutputEIEO(command, errfatal)
+    return output
+
+# Return the contents of the file. CFE = Checking For Error.
+# Report an error and proceed if something went wrong.
+def fileContentsCFE(filename):
+    p = subprocess.Popen(['cat', filename],
```

Comment:
If you're going to go through this effort, we should stop shelling out to call `cat`

We should just read the files contents in pure python